src/anniversary/sf33rd/Source/Game/bg.c

```diff
@@ -61,17 +260,153 @@ INCLUDE_RODATA("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", literal_502
+            x += 128;
+        }
+        y += 128;
+    }
+}
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", bgAkebonoDraw);
+void bgDrawOneChip(s32 x, s32 y, s32 xs, s32 ys, s32 gbix, u32 vtxCol, s32 ofsPal) {
+    s32 i;
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", ppgCalScrPosition);
+    if (No_Trans == 0 && ppgCheckTextureNumber(0, gbix)) {
+        ppgCalScrPosition(x, y, xs, ys);
+        if (!(scrDrawPos->x < 384.0f && !(scrDrawPos[3].x < 0.0f) && (scrDrawPos->y < 224.0f) &&
+              !(scrDrawPos[3].y < 0.0f))) {
+            return;
+        } else {
+            ppgWriteQuadUseTrans(scrDrawPos, vtxCol, 0, gbix, 0, 0, ofsPal);
+        }
+    }
+}
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/bg", scr_trans_sub2);
+void bgAkebonoDraw() {
+    s32 i;
+
+    scrDrawPos->x = 0.0f;
+    scrDrawPos->y = 0.0f;
+    scrDrawPos[3].x = 128.0f;
+    scrDrawPos[3].y = 224.0f;
+    scrDrawPos->z = scrDrawPos[3].z = *(&PrioBase[bg_priority[3]]);
+    scrDrawPos->s = scrDrawPos->t = 0.0f;
+    scrDrawPos[3].s = 1.0f;
+    scrDrawPos[3].t = 0.875f;
+    i = 0;
+
+    while (i < 3) {
+        ppgWriteQuadUseTrans(scrDrawPos, -1U, NULL, i, i, 0, 0);
+        scrDrawPos->x += 128.0f;
+        scrDrawPos[3].x += 128.0f;
+        i++;
+    }
+}
+
+void ppgCalScrPosition(s32 x, s32 y, s32 xs, s32 ys) {
+    Vec3 point[2];
+
+    point[0].x = (f32)x;
+    point[0].y = (f32)y;
+    point[1].x = (f32)(x + xs);
+    point[1].y = (f32)(y + ys);
+    point[0].z = point[1].z = 0;
+    njCalcPoints(0, &point[0], &point[0], 2);
+    scrDrawPos[0].x = scrDrawPos[2].x = point[0].x;
+    scrDrawPos[0].y = scrDrawPos[1].y = point[0].y;
+    scrDrawPos[1].x = scrDrawPos[3].x = point[1].x;
+    scrDrawPos[2].y = scrDrawPos[3].y = point[1].y;
+    scrDrawPos[0].z = scrDrawPos[1].z = scrDrawPos[2].z = scrDrawPos[3].z = point[0].z;
+    scrDrawPos[0].s = (0.5f + (f32)(x & 0x7F)) / 128.0f;
+    scrDrawPos[0].t = (0.5f + (f32)(y & 0x7F)) / 128.0f;
+    scrDrawPos[3].s = (0.5f + (f32)((x & 0x7F) + xs)) / 128.0f;
+    scrDrawPos[3].t = (0.5f + (f32)((y & 0x7F) + ys)) / 128.0f;
+    scrDrawPos[1].s = scrDrawPos[3].s;
+    scrDrawPos[2].s = scrDrawPos[0].s;
+    scrDrawPos[1].t = scrDrawPos[0].t;
+    scrDrawPos[2].t = scrDrawPos[3].t;
+}
+
+void scr_trans_sub2(s32 x, s32 y, s32 suzi) {
+    Vec3 point[2];
+    Vec3 spoint[2];
+
+    point[0].x = (f32)x;
+    spoint[0].x = (f32)(x + suzi);
+    point[0].y = spoint[0].y = (f32)(y + 0x200);
+    point[1].x = (f32)(x + 0x100);
+    spoint[1].x = (f32)(x + suzi + 0x100);
+    point[1].y = spoint[1].y = (f32)(y + 0x300);
+    point[0].z = point[1].z = spoint[0].z = spoint[1].z = 0;
+    njCalcPoints(NULL, &point[0], &point[0], 2);
+    njCalcPoints(NULL, &spoint[0], &spoint[0], 2);
+    bgpoly[0].x = spoint[0].x;
+    bgpoly[0].y = point[0].y;
+    bgpoly[0].z = point[0].z;
+    bgpoly[0].u = 0.0f;
+    bgpoly[0].v = 0.0f;
+    bgpoly[0].col = -1;
```

Comment:
Colors should be written as hex. They are ARBG values after all
```suggestion
    bgpoly[0].col = 0xFFFFFFFF;
```